core/jobs/src/test/java/com/cognifide/aet/job/common/modifiers/replacetext/ReplaceTextModifierTest.java

```diff
@@ -0,0 +1,146 @@
+ * limitations under the License.
+ */
+package com.cognifide.aet.job.common.modifiers.replacetext;
+
+import com.cognifide.aet.job.api.collector.CollectorProperties;
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import com.cognifide.aet.job.api.exceptions.ProcessingException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+import java.util.Map;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ReplaceTextModifierTest {
+
+  private static final String PARAM_XPATH = "xpath";
+  private static final String PARAM_CSS = "css";
+  private static final String ATTRIBUTE_PARAM = "attributeName";
+  private static final String VALUE_PARAM = "value";
+
+  private static final String PARAM_XPATH_VALUE = "//*[@id='toRemove']";
+  private static final String PARAM_CSS_VALUE = "@logo > a";
+  private static final String PARAM_ATTRIBUTE_VALUE = "href";
+  private static final String PARAM_VALUE_VALUE = "aetaetaet";
+
+  private static final String URL = "http://www.cognifide.com";
+
+  @Mock
+  private WebDriver webDriver;
+
+  @Mock
+  private CollectorProperties properties;
+
+  @InjectMocks
+  private ReplaceTextModifier tested;
+
+  @Mock
+  private Map<String, String> params;
+
+  @Before
+  public void setUp() {
+    when(webDriver.getCurrentUrl()).thenReturn(URL);
+  }
+
+  @Test
+  public void setParameters_XPathIsValid_ValidationPassedSuccessfuly() throws ParametersException {
+    when(params.containsKey(PARAM_XPATH)).thenReturn(true);
+    when(params.get(PARAM_XPATH)).thenReturn(PARAM_XPATH_VALUE);
+    when(properties.getUrl()).thenReturn(URL);
+    tested.setParameters(params);
+  }
+
+  @Test
+  public void setParameters_CssIsValid_ValidationPassedSuccessfuly() throws ParametersException {
+    when(params.containsKey(PARAM_CSS)).thenReturn(true);
+    when(params.get(PARAM_CSS)).thenReturn(PARAM_CSS_VALUE);
+    when(properties.getUrl()).thenReturn(URL);
+    tested.setParameters(params);
+  }
+
+  @Test(expected = ParametersException.class)
+  public void setParameters_CssAndXpathArePassed_ExceptionIsThrown() throws ParametersException {
+    when(params.containsKey(PARAM_CSS)).thenReturn(true);
+    when(params.get(PARAM_CSS)).thenReturn(PARAM_CSS_VALUE);
+
+    when(params.containsKey(PARAM_XPATH)).thenReturn(true);
+    when(params.get(PARAM_XPATH)).thenReturn(PARAM_XPATH_VALUE);
+
+    when(properties.getUrl()).thenReturn(URL);
+    tested.setParameters(params);
+  }
+
+  @Test(expected = ParametersException.class)
+  public void setParameters_CssAndXpathAreNotPassed_ExceptionIsThrown() throws ParametersException {
+    when(properties.getUrl()).thenReturn(URL);
```

Comment:
Do we need this `properties` mock? If not please remove it. If yes, please make it more clear what is the purpose of this mock. (must have).